qwerty_ast/src/ast.rs

```diff
@@ -349,6 +349,74 @@ impl<E> FunctionDef<E> {
     }
 }
 
+// NOTE: Until there is a nice way to distinguish these, there
+// are two display implementations. One could use Trivializable,
+// but you would have to modify it to return Option<Expr> and
+// this seemed antithetical to the purpose. I also didn't want
+// to create another trait just for this.
+impl std::fmt::Display for FunctionDef<qpu::Expr> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "fn {}(", self.name)?;
+        for (i, (ty, arg_name)) in self.args.iter().enumerate() {
+            write!(f, "{arg_name} {ty}")?;
+            if i + 1 != self.args.len() {
+                write!(f, ", ")?;
+            }
+        }
+        write!(f, ") -> {}", self.ret_type)?;
+        if self.is_rev {
+            write!(f, " rev")?;
```

Comment:
(If we use Python DSL syntax, this would be a `@reversible` decorator before the function)